### DISCUSSION TITLE
Replacing textures for loaded models

### DISCUSSION THREAD

**Discussion by m-paunovic-vegait** [sceneview/sceneview#110] [Q&A]
Hi! What is the proper procedure for loading in a new image to use as a texture for loaded models? I have an already-textured model, and an image, and I would like to replace the current texture with the new image. I am able to get references to the existing textures by name.

I tried using TextureLoader to load the texture (which seems to work) and call setTexture on the model instance to apply it (like I've seen it done with the plane visualizer), but I'm getting an error saying the sampler couldn't be found and the app crashes shortly after.

Thanks in advance for your help!

**Comment by ThomasGorisse** [comment]
Hi!
You can have a look at the not yet moved to SceneView [Image Texture Sample from Sceneform](https://github.com/SceneView/sceneform-android/tree/master/samples/image-texture)

**Reply by m-paunovic-vegait** [reply]
Regarding the TextureBuilder, it returns a com.google.ar.sceneform.rendering.Texture, while the setTexture expects a com.google.android.filament.Texture. Did I import the wrong thing by accident? I tried a cast and it didn't seem to work

**Comment by ThomasGorisse** [comment]
You shouldn't use the old `TextureBuilder` anymore but the Filament `Texture.Builder()` one directly.
You can have a look at https://github.com/SceneView/sceneview-android/blob/f7247805f720ab7842ffd463bd23bff5f853fedc/sceneview/src/main/java/io/github/sceneview/texture/TextureLoader.kt#L15-L83 for some helpers.

**Reply by m-paunovic-vegait** [reply]
Figured it out! My first attempt was pretty close, just needed to use the "baseColorMap" as the name. Thank you for your assistance and for your patience with me!
